### PR TITLE
Getting rid of abstract types and using new version of Lego

### PR DESCRIPTION
**Pull request by shrey-garg**

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -35,6 +35,11 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
+            <version>1.10.19</version>
```
Comment (pmohankumar) [inline api/pom.xml]: Why is this added?

**Comment by pmohankumar** [comment]
Please update migration wiki here
https://github.com/flipkart-incubator/Poseidon/wiki/Migration-Guide-%5B4.x-to-5.x%5D

File: api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java [hunk 2]
```diff
@@ -99,12 +99,13 @@ public Buildable getBuildable(Request request) throws LegoSetException, ElementN
...
             RequestContext.set(ENDPOINT_NAME, pojo.getName());
-        }
 
-        String name = buildable.getName();
-        if (name != null && !name.isEmpty()) {
-            poseidonRequest.setAttribute(TIMER_CONTEXT, Metrics.getRegistry().timer("poseidon.api." + name + "." + httpMethod).time());
+            String name = pojo.getName();
+            if (name != null && !name.isEmpty()) {
+                poseidonRequest.setAttribute(TIMER_CONTEXT, Metrics.getRegistry().timer("poseidon.api." + name + "." + httpMethod).time());
+            }
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java]: Alignment issue here? I see buildable.getName() changed to pojo.getName() but whole block is showing as changed

File: api/pom.xml (hunk 1 above)
Comment (shrey-garg) [inline api/pom.xml]: tests in api module fail for me otherwise..

File: api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java (hunk 2 above)
Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java]: This block has been moved inside the if condition above

File: api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java [hunk 3]
```diff
@@ -30,13 +30,4 @@ public ListDataType(Collection<? extends T> collection) {
         super(collection);
     }
 
-    @Override
-    public String getShortDescription() {
```
Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: Done

**Comment by shrey-garg** [comment]
Updated the migration guide

**Comment by pmohankumar** [comment]
Can you resolve the merge conflicts?

File: api/src/main/java/com/flipkart/poseidon/internal/OrchestratorDataSource.java [hunk 4]
```diff
@@ -67,14 +67,4 @@ public DataType call() throws Exception {
...
-
-    @Override
-    public String getShortDescription() {
-        return "Internal DataSource that acts as an Orchestrator for Hydra based dispatcher-composer tasks.";
-    }
-
-    @Override
-    public String getDescription() {
-        return getShortDescription();
-    }
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/internal/OrchestratorDataSource.java]: Please use annotations for these removed descriptions

File: core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java [hunk 5]
```diff
@@ -51,42 +46,13 @@ public AbstractDataSource(LegoSet legoset, Request request) {
...
-    @Override
-    public String getShortDescription() {
-        return this.getClass().getSimpleName();
-    }
-
-    @Override
-    public String getDescription() {
-        return this.getClass().getName();
-    }
-
```
Comment (pmohankumar) [inline core/src/main/java/com/flipkart/poseidon/datasources/AbstractDataSource.java]: Please use annotations for these removed descriptions

**Comment by pmohankumar** [comment]
Good to see this cleaned up. Thanks @shrey-garg
